curvefs/src/client/dentry_cache_manager.cpp

```diff
@@ -61,11 +59,6 @@ CURVEFS_ERROR DentryCacheManagerImpl::GetDentry(uint64_t parent,
             return CURVEFS_ERROR::OK;
         }
     }
-    curve::common::WriteLockGuard lg(mtx_);
-    bool ok = dCache_->Get(key, out);
-    if (ok) {
-        return CURVEFS_ERROR::OK;
-    }
 
     MetaStatusCode ret = metaClient_->GetDentry(fsId_, parent, name, out);
```

Comment:
if don‘t lock GetDentry here， when concurrent with InsertOrReplaceCache above， this fuction maybe return old version of dentry？

Reply:
we need get from cache first, if cache do not have this one, then  get from metaserver.

may be readlock can be use by get.

Reply:
Using namelock to lock the specific one. If lock entire icache will block requests to other inode.

Reply:
> if don‘t lock GetDentry here， when concurrent with InsertOrReplaceCache above， this fuction maybe return old version of dentry？

Using namelock to lock the specific one. If lock entire icache will block requests to other inode.

Reply:
> we need get from cache first, if cache do not have this one, then get from metaserver.
> 
> may be readlock can be use by get.

The key issue is the lock for dCache will block requests on other dentry. Using namelock to lock the specific one.